packages/jest-cli/src/snapshot_interactive_mode.js

```diff
@@ -13,68 +13,163 @@ import type {AggregatedResult} from 'types/TestResult';
 const chalk = require('chalk');
 const ansiEscapes = require('ansi-escapes');
 const {pluralize} = require('./reporters/utils');
-const {KEYS} = require('./constants');
+const {KEYS, ARROW} = require('./constants');
 
 export default class SnapshotInteractiveMode {
   _pipe: stream$Writable | tty$WriteStream;
   _isActive: boolean;
-  _updateTestRunnerConfig: (path: string, shouldUpdateSnapshot: boolean) => *;
-  _testFilePaths: Array<string>;
+  _updateTestRunnerConfig: (
+    assertion: ?Object,
+    shouldUpdateSnapshot: boolean,
+  ) => *;
+  _testAssertions: Array<Object>;
```

Comment:
Can we strengthen this type to something like `type TestAssertion = {path: string, title: string}` and reuse it here https://github.com/facebook/jest/pull/5864/files#diff-f6e31ceb30acd38abf7312e70f26edd6R17?

Reply:
Sure can - would you put this in the [TestResult](https://github.com/facebook/jest/blob/master/types/TestResult.js) types or keep it local?

Reply:
I'd put it there ;)

Reply:
Done, but naming is hardI called it AssertionLocation as the path + title pair are used to identify the spec location. Note this is not unique since multiple tests in the same file can have the same title. If that happens then the interactive mode will display and update/skip both snapshots

Reply:
Yea, I'm not sure about the naming either, but can't come up with anything better 😅